pkg/types/config/validate.go

```diff
@@ -245,7 +244,7 @@ func (c *Cluster) validateS3Bucket() error {
 		return fmt.Errorf("the S3 bucket name %q, generated from the cluster name and base domain, is too long; S3 bucket names must be less than 63 characters; please choose a shorter cluster name or base domain", bucket)
 	}
 	if !regexp.MustCompile("^[a-z0-9][a-z0-9-.]{1,61}[a-z0-9]$").MatchString(bucket) {
-		return errors.New("invalid characters in S3 bucket name")
+		return fmt.Errorf("invalid characters in S3 bucket name %q, generated from the cluster name and base domain", bucket)
```

Comment:
This is an improvement on the error message. However, if we can, we should improve the validation of the user input earlier to catch invalid bucket names. The current validation of the base domain currently allows a trailing dot. If a trailing dot results in an invalid S3 bucket name later, then either (1) we need to reject a base domain with a trailing dot or (2) we need to accommodate the trailing dot to make a valid S3 bucket name.

Reply:
> This is an improvement on the error message. However...

@dak1n1, let me know if you want to pick these suggestions up or punt them to futute work.  Theyre good suggestions, but I'm fine getting there in baby steps.

Reply:
@wking I like the idea of catching the error earlier. I'll be in the SFO office next week. This could be a good task for me to work on, since this is my very first golang commit.